libazureinit/src/user.rs

```diff
@@ -1,95 +1,177 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-use std::fs;
-use std::fs::create_dir;
-use std::fs::File;
-use std::io::Write;
+use std::{
+    fs::Permissions,
+    io::Write,
+    os::unix::fs::{DirBuilderExt, PermissionsExt},
+};
 
-use nix::unistd::{Gid, Uid};
-use std::ffi::CString;
-use std::os::unix::fs::PermissionsExt;
+use tracing::instrument;
 
 use crate::error::Error;
 use crate::imds::PublicKeys;
 
-pub async fn set_ssh_keys(
+pub fn set_ssh_keys(
     keys: Vec<PublicKeys>,
-    username: String,
-    file_path: String,
+    username: impl AsRef<str>,
 ) -> Result<(), Error> {
-    let mut authorized_keys_path = file_path;
-    authorized_keys_path.push_str("/authorized_keys");
-
-    let mut authorized_keys = File::create(authorized_keys_path.clone())?;
-    for key in keys {
-        writeln!(authorized_keys, "{}", key.key_data)?;
-    }
-    let metadata = fs::metadata(authorized_keys_path.clone())?;
-    let permissions = metadata.permissions();
-    let mut new_permissions = permissions;
-    new_permissions.set_mode(0o600);
-    fs::set_permissions(authorized_keys_path.clone(), new_permissions)?;
-
-    let uid_username = CString::new(username.clone())?;
-    let uid_passwd = unsafe { libc::getpwnam(uid_username.as_ptr()) };
-    let uid = unsafe { (*uid_passwd).pw_uid };
-    let new_uid = Uid::from_raw(uid);
-
-    let gid_groupname = CString::new(username)?;
-    let gid_group = unsafe { libc::getgrnam(gid_groupname.as_ptr()) };
```

Comment:
Removing the libc-related code like that, it becomes now unnecessary to keep `libc` any more.
Can you please remove [libc](https://github.com/Azure/azure-init/blob/ffe6d547847e56341137d45a01a20456783f2353/libazureinit/Cargo.toml#L22) in `libazureinit/Cargo.toml`?

To catch such issues, we should actually regularly run `cargo machete` or so. And there are other unused crates in this repo, but I will soon create another PR to remove those.

Reply:
Good catch, I've dropped it